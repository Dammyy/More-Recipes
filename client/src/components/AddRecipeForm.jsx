import React, { PureComponent } from 'react';
import { Link } from 'react-router';
import { Field, reduxForm } from 'redux-form/immutable';
import PropTypes from 'prop-types';
/**
 *
 *
 * @export
 * @class AddRecipeForm
 * @extends {PureComponent}
 */
class AddRecipeForm extends PureComponent {
  /**
   *
   *
   *  @param {any} event
   * @returns {string} redirect
   * @memberOf AddRecipe
   */
  /** html component to render
   *
   *
   * @returns {void}
   *
   * @memberOf AddRecipeForm
   */
  render() {
    const { image, uploadImage } = this.props;
    return (
      <div className="container add-recipe-page">
        <div className="col-md-12 latest-recipes">
          <div className="text-left btn-buttons">
            <Link
              to="/catalog/manage"
              className="btn btn-info"
            >
              Back to My Recipes
            </Link>
          </div>
          <div className="panel panel-default">
            <div className="panel-heading">
              <h2 className="panel-title text-center">
                Add a Recipe!
              </h2>
            </div>
            <div className="panel-body">
              <form
                name="form"
                action=""
                onSubmit={this.props.handleSubmit}
                noValidate
              >
                <div className="form-group">
                  <label htmlFor="Title">Title</label>
                  <Field
                    name="title"
                    type="text"
                    id="Title"
                    component="input"
                    className="form-control"
                    placeholder="Title"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="ingredients">Ingredients</label>
                  <Field
                    name="ingredients"
                    component="input"
                    className="form-control"
                    placeholder="Ingredients e.g water, potato, rice"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="price">Details</label>
                  <Field
                    name="details"
                    component="textarea"
                    className="form-control"
                    rows="5"
                    placeholder="Recipe details"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="image">Image</label>
                  <div className="text-center dropup">
                    <button
                      id="button-upload"
                      type="button"
                      className="btn btn-danger"
                      onClick={() => uploadImage()}
                    >
                      Upload <span className="caret" />
                    </button>
                  </div>
                </div>
                <div className="form-group text-center upload-image">
                  <img
                    id="image"
                    className="img-responsive img-upload"
                    alt=""
                    src={image}
                  />
                </div>
                <button
                  type="submit"
                  className="btn btn-primary btn-block"
                >
                  Publish
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

AddRecipeForm.propTypes = {
  uploadImage: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  image: PropTypes.string.isRequired
};

export default reduxForm({ form: 'recipe' })(AddRecipeForm);
