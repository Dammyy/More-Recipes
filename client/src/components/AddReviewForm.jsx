import React, { PureComponent } from 'react';
import { Field, reset, reduxForm } from 'redux-form/immutable';
import PropTypes from 'prop-types';
/**
 *
 *
 * @export
 * @class AddRecipeForm
 * @extends {PureComponent}
 */
class AddReviewForm extends PureComponent {
  /** html component to render
   *
   *
   * @returns {void}
   *
   * @memberOf AddReviewForm
   */
  render() {
    return (

      <div id="recipe-display" className="col-md-12">
        <h2 className="panel-title text-center r-d-titles">
              Leave a review
        </h2>
        <div className="panel-body">
          <form
            name="form"
            action=""
            onSubmit={this.props.handleSubmit}
            noValidate
          >
            <div className="form-group">
              <Field
                name="review"
                type="text"
                id="review-input-field"
                component="textarea"
                className="form-control"
                placeholder="Leave a Review"
              />
            </div>
            <button
              type="submit"
              className="btn btn-primary btn-block"
            >
                Submit
            </button>
          </form>
        </div>
      </div>
    );
  }
}

AddReviewForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

const afterSubmit = (result, dispatch) =>
  dispatch(reset('review'));

export default reduxForm({
  form: 'review',
  onSubmit: afterSubmit,
})(AddReviewForm);
