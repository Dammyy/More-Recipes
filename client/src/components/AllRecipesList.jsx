import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import RecipeHome from './RecipeHome';
import {
  BtnHome,
  AuthenticatedBtnAdd,
  AuthenticatedBtnManageRecipes,
  AuthenticatedBtnFavorites,
  AuthenticatedBtnProfile
} from './Buttons';

const BtnCurrent = (() => (
  'Recipes Catalog'
));

/**
 *
 *
 * @class AllRecipesList
 * @extends {PureComponent}
 */
class AllRecipesList extends PureComponent {
  /**
   * @memberOf AllRecipesList
   * @returns {void}
   */
  render() {
    const {
      recipes,
    } = this.props;
    return (
      <div>
        <div className="text-left-buttons btn-buttons">
          <BtnHome />
          <AuthenticatedBtnAdd />
          <AuthenticatedBtnManageRecipes />
          <AuthenticatedBtnFavorites />
          <AuthenticatedBtnProfile />
        </div>
        <div className="current-page btn-buttons"><BtnCurrent /></div>
        <div className="col-md-12 latest-recipes">
          <div className="row">
            {
              recipes && recipes.map((recipe) => {
                return (
                  <RecipeHome
                    {...recipe}
                    key={recipe.id}
                  />
                );
              })
            }
          </div>
        </div>
      </div>
    );
  }
}
AllRecipesList.propTypes = {
  recipes: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
};

export default AllRecipesList;
