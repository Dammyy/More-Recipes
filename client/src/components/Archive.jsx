import React, { PureComponent } from 'react';
import { Link } from 'react-router';

export default class Home extends PureComponent {
  active (path) {
    if (this.props.location.pathname === path) {
      return 'active';
    }
  }
  render () {
    return (
<div>
        <nav className="navbar navbar-toggleable-md navbar-expand-md navbar-light bg-faded">
          <div className="header-logo">
            <a className="navbar-brand" href="index.html">
              <img src="img/logo.jpg" width={50} height={50} className="d-inline-block align-top" alt />
              <b>More Recipes<b>
                </b></b></a><b><b>
              </b></b></div><b><b>
              <button className="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon" />
              </button>
              <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                  <li className="nav-item">
                    <a className="nav-link" href="catalog.html">Catalog</a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="fav.html">Favorites</a>
                  </li>
                  <li className="nav-item" className={this.active('/manage')}>
                    <Link to="/manage" className="nav-link">Manage recipes</Link>
                  </li>
                  <li className="nav-item" className={this.active('/catalog')}>
                    <Link to="/catalog" className="nav-link">Catalog</Link>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="profile.html">Profile</a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="signup.html">Sign up</a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="signin.html">Sign in</a>
                  </li>
                </ul>
				
              </div>
            </b></b></nav><b><b>
            <div className="container-fluid">
              <div className="row">
                <div className="col-md-9 catalog-left">
                  <div id="catalog-search-form">
                    <form className="form-inline">
                      <input className="form-control mr-sm-2" type="text" placeholder="Search" />
                      <button className="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
                    </form> 
                  </div>
                  <div className="col-md-12 latest-recipes">
                  <div className="row">
                    {this.props.children}
                </div>
                </div>
                </div>
                <div className="col-md-3 catalog-right">
                  <h3 id="latest-h3">Popular Recipes</h3>
                  <div id="popular">
                    <div id="popular-title"><a href="#"> Awesome chicken </a></div>
                    <div id="popular-votes"><li><i className="fa fa-thumbs-o-up" aria-hidden="true"> 200</i> </li><li><i className="fa fa-thumbs-o-down" aria-hidden="true"> 50</i></li> </div>
                  </div>
                  <div id="popular">
                    <div id="popular-title"><a href="#"> Grilled turkey </a></div>
                    <div id="popular-votes"><li><i className="fa fa-thumbs-o-up" aria-hidden="true"> 200</i> </li><li><i className="fa fa-thumbs-o-down" aria-hidden="true"> 50</i></li> </div>
                  </div>
                  <div id="popular">
                    <div id="popular-title"><a href="#"> Awesome chicken </a></div>
                    <div id="popular-votes"><li><i className="fa fa-thumbs-o-up" aria-hidden="true"> 200</i> </li><li><i className="fa fa-thumbs-o-down" aria-hidden="true"> 50</i></li> </div>
                  </div>
                </div>
              </div>
            </div>


            <footer className="footer">
              <div className="container">
                <div className="row">
                  <div className="col-md-6">
                    <a href="https://github.com/Dammyy"> <img src="img/github.png" /></a>
                  </div>
                  <div className="col-md-6">
                    <span className="text-muted">Copyright Â© More Recipes!</span>
                  </div>
                </div>
              </div>
            </footer>
            
          </b></b></div>
          
    );
  }
}
