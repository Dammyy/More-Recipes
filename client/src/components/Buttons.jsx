import React from 'react';
import { Link } from 'react-router';
import UserIsAuthenticated from '../utils/authWrapper';

const options = {
  authSelector: state => state.get('auth'),
  predicate: auth => auth.get('Authenticated'),
  wrapperDisplayName: 'authRecipeButtons',
  FailureComponent: null
};

const BtnAdd = UserIsAuthenticated(options)(() => (
  <Link
    to="/catalog/add"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-pencil-square-o" aria-hidden="true" /> Add Recipe
  </Link>
));
const BtnManageRecipes = UserIsAuthenticated(options)(() => (
  <Link
    to="/catalog/manage"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-list-alt" aria-hidden="true" /> Manage Recipes
  </Link>
));

const BtnFavorites = UserIsAuthenticated(options)(() => (
  <Link
    to="/catalog/favorites"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-heart" /> My Favorites
  </Link>
));

const BtnHome = (() => (
  <Link
    to="/"
    className="btn btn-info btn-manage"
  >
    <i className="fa fa-home" /> Home
  </Link>
));

const BtnCatalog = (() => (
  <Link
    to="/catalog"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-list-alt" aria-hidden="true" /> Catalog
  </Link>
));

const BtnProfile = UserIsAuthenticated(options)(() => (
  <Link
    to="/profile"
    className="btn btn-publish btn-manage"
  >
    <i className="fa fa-user-o" aria-hidden="true" /> My Profile
  </Link>
));

export {
  BtnAdd,
  BtnManageRecipes,
  BtnHome,
  BtnCatalog,
  BtnFavorites,
  BtnProfile
};
