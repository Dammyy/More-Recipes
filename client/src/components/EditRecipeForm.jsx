import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { Field, reduxForm } from 'redux-form/immutable';
import PropTypes from 'prop-types';
import {
  BtnHome,
  AuthenticatedBtnAdd,
  AuthenticatedBtnManageRecipes,
  AuthenticatedBtnFavorites,
  AuthenticatedBtnProfile,
  BtnCatalog,
  BtnCancel
} from './Buttons';

const BtnCurrent = (() => (
  'UPDATE RECIPE'
));
/**
 *
 *
 *
 * @class EditRecipeForm
 * @extends {PureComponent}
 */
class EditRecipeForm extends PureComponent {
  /**
   *
   * @memberOf EditRecipeForm
   * @returns {void}
   *
   */
  componentDidMount() {
    if (!this.props.recipes) {
      this.props.redirectUser('/catalog');
    }
  }


  /**
   *
   *
   * @memberOf EditRecipeForm
   * @returns {void}
   */
  render() {
    const { image, uploadImage } = this.props;
    return (
      <div>
        <div className="text-left-buttons btn-buttons">
          <BtnHome />
          <AuthenticatedBtnAdd />
          <BtnCatalog />
          <AuthenticatedBtnManageRecipes />
          <AuthenticatedBtnFavorites />
          <AuthenticatedBtnProfile />
          <Link
            className="btn btn-primary"
            to={`view/${this.props.id}`}
            params={{ id: this.props.id }}
          >
          View
          </Link >
          <BtnCancel />
        </div>
        <div className="current-page btn-buttons"><BtnCurrent /></div>
        <div className="container add-recipe-page">
          <div className="col-md-12 latest-recipes">
            <div className="panel-body">
              <form
                name="form"
                action=""
                onSubmit={this.props.handleSubmit}
                noValidate
              >
                <div className="form-group">
                  <label htmlFor="Title">Title</label>
                  <Field
                    name="title"
                    type="text"
                    id="Title"
                    component="input"
                    className="form-control"
                    placeholder="Title"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="ingredients">Ingredients</label>
                  <Field
                    name="ingredients"
                    component="input"
                    className="form-control"
                    placeholder="Ingredients e.g water, potato, rice"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="price">Details</label>
                  <Field
                    name="details"
                    component="textarea"
                    className="form-control"
                    rows="5"
                    placeholder="Recipe details"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="image">Image</label>
                  <div className="text-center dropup">
                    <button
                      id="button-upload"
                      type="button"
                      className="btn btn-danger"
                      onClick={() => uploadImage()}
                    >
                    Upload <span className="caret" />
                    </button>
                  </div>
                </div>
                <div className="form-group text-center">
                  <img
                    id="image"
                    className="img-responsive img-upload"
                    alt=""
                    src={image}
                  />
                </div>
                <button
                  type="submit"
                  className="btn btn-primary btn-block"
                >
                Publish
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

EditRecipeForm.propTypes = {
  uploadImage: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  image: PropTypes.string.isRequired,
  recipes: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
  redirectUser: PropTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => {
  const { recipes, id } = ownProps;
  const recipe = recipes.recipes
    .filter(recp => recp.id === parseInt(id, 10))[0];
  return {
    initialValues: recipe,
  };
};

const connectReduxForm = reduxForm({
  form: 'updateRecipe',
  enableReinitialize: true,
})(EditRecipeForm);

export default connect(mapStateToProps)(connectReduxForm);
