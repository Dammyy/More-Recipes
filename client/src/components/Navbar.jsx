import React, { Component } from 'react';
import { Link } from 'react-router';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form/immutable';
import * as recipesActionCreators from '../actions/recipes';
/**
 *
 * @class Navbar
 * @extends {Component}
 */
export class Navbar extends Component {
  /**
   * Creates an instance of Navbar.
   * @memberOf Navbar
   * @param {any} props
   */
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  /**
   * @memberOf Navbar
   * @returns {void}
   * @param {any} event
   */
  handleSubmit(event) {
    event.preventDefault();
    this.props.recipesActions.searchRecipes();
  }

  /**
   *
   * @memberOf Navbar
   * @returns {void}
   *
   */
  render() {
    return (
      <nav className="navbar navbar-expand-md navbar-light">
        <div className="logo-link-wrap">
          <Link to="/" className="nav-link header-logo">
            <img
              src="img/logo.jpg"
              width={50}
              height={50}
              className="navbar-brand"
              alt="logo"
            />
          More Recipes
          </Link>
        </div>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarsExampleDefault"
          aria-controls="navbarsExampleDefault"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarsExampleDefault">
          <ul className="navbar-nav mr-auto" />
          <form
            className="form-inline my-2 my-md-0"
            onSubmit={this.handleSubmit}
          >
            <Field
              name="search"
              className="form-control"
              component="input"
              placeholder="Search"
            />
            <button
              className="btn btn-search my-2 my-sm-0"
              type="submit"
            ><i className="fa fa-search" aria-hidden="true" /> Search
            </button>
          </form>
        </div>
      </nav>
    );
  }
}

/**
 *
 * @param {any} dispatch
 * @returns {void}
 */
export function mapDispatchToProps(dispatch) {
  return {
    recipesActions: bindActionCreators(recipesActionCreators, dispatch)
  };
}

Navbar.propTypes = {
  recipesActions: PropTypes.objectOf(PropTypes.func).isRequired,
};

export default
reduxForm({ form: 'search' })(connect(null, mapDispatchToProps)(Navbar));
