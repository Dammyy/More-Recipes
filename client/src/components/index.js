import Home from './Home';
import Welcome from './Welcome';
import Archive from './Archive';
import AddRecipeForm from './AddRecipeForm';
import EditRecipeForm from './EditRecipeForm';
import Recipe from './Recipe';
import RecipesList from './RecipesList';
import Login from './Login';
import Signup from './Signup';
import ViewRecipe from './ViewRecipe';
import NotFoundPage from './NotFoundPage';
import AllRecipesList from './AllRecipesList';
import MyFavoriteRecipes from './MyFavoriteRecipes';
import AddReviewForm from './AddReviewForm';
import Reviews from './Reviews';

export { Home, Welcome, AddRecipeForm, Recipe,
  RecipesList, Archive, Login, Signup, EditRecipeForm,
  ViewRecipe, NotFoundPage, AllRecipesList, MyFavoriteRecipes,
  AddReviewForm, Reviews };
