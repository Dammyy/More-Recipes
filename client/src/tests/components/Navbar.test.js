import React from 'react';
import { shallow } from 'enzyme';
import { Navbar, mapDispatchToProps } from '../../components/Navbar';

describe('Navbar component', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const component = shallow(<Navbar />);
      expect(component).toMatchSnapshot();
    });
  });
});

describe('Instance methods', () => {
  describe('Navbar instance methods', () => {
    test('', () => {
      const props = {
        recipesActions: { searchRecipes: jest.fn(), },
        handleSubmit: jest.fn(),
      };
      const component = shallow(<Navbar {...props} />);
      component.instance().handleSubmit({ preventDefault: () => {} });
      expect(props.recipesActions.searchRecipes).toBeCalled();
    });
  });
});
describe('mapDispatchToProps', () => {
  it('', () => {
    const dispatch = jest.fn();
    const mdtp = mapDispatchToProps(dispatch);
    expect(mdtp).toHaveProperty('recipesActions');
  });
});
