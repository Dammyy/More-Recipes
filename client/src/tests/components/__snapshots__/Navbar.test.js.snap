// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo-link-wrap"
        >
          <Link
            className="nav-link header-logo"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            <img
              alt="logo"
              className="navbar-brand"
              height={50}
              src="img/logo.jpg"
              width={50}
            />
            More Recipes
          </Link>
        </div>,
        <button
          aria-controls="navbarsExampleDefault"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarsExampleDefault"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>,
        <div
          className="collapse navbar-collapse"
          id="navbarsExampleDefault"
        >
          <ul
            className="navbar-nav mr-auto"
          />
          <form
            className="form-inline my-2 my-md-0"
            onSubmit={[Function]}
          >
            <Field
              className="form-control"
              component="input"
              name="search"
              placeholder="Search"
            />
            <button
              className="btn btn-search my-2 my-sm-0"
              type="submit"
            >
              <i
                aria-hidden="true"
                className="fa fa-search"
              />
               Search
            </button>
          </form>
        </div>,
      ],
      "className": "navbar navbar-expand-md navbar-light",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="nav-link header-logo"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/"
          >
            <img
              alt="logo"
              className="navbar-brand"
              height={50}
              src="img/logo.jpg"
              width={50}
            />
            More Recipes
          </Link>,
          "className": "logo-link-wrap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="navbar-brand"
                height={50}
                src="img/logo.jpg"
                width={50}
              />,
              "More Recipes",
            ],
            "className": "nav-link header-logo",
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "navbar-brand",
                "height": 50,
                "src": "img/logo.jpg",
                "width": 50,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "More Recipes",
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-controls": "navbarsExampleDefault",
          "aria-expanded": "false",
          "aria-label": "Toggle navigation",
          "children": <span
            className="navbar-toggler-icon"
          />,
          "className": "navbar-toggler",
          "data-target": "#navbarsExampleDefault",
          "data-toggle": "collapse",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "navbar-toggler-icon",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              className="navbar-nav mr-auto"
            />,
            <form
              className="form-inline my-2 my-md-0"
              onSubmit={[Function]}
            >
              <Field
                className="form-control"
                component="input"
                name="search"
                placeholder="Search"
              />
              <button
                className="btn btn-search my-2 my-sm-0"
                type="submit"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-search"
                />
                 Search
              </button>
            </form>,
          ],
          "className": "collapse navbar-collapse",
          "id": "navbarsExampleDefault",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navbar-nav mr-auto",
            },
            "ref": null,
            "rendered": null,
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Field
                  className="form-control"
                  component="input"
                  name="search"
                  placeholder="Search"
                />,
                <button
                  className="btn btn-search my-2 my-sm-0"
                  type="submit"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-search"
                  />
                   Search
                </button>,
              ],
              "className": "form-inline my-2 my-md-0",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "form-control",
                  "component": "input",
                  "name": "search",
                  "placeholder": "Search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      aria-hidden="true"
                      className="fa fa-search"
                    />,
                    " Search",
                  ],
                  "className": "btn btn-search my-2 my-sm-0",
                  "type": "submit",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-search",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " Search",
                ],
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo-link-wrap"
          >
            <Link
              className="nav-link header-logo"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/"
            >
              <img
                alt="logo"
                className="navbar-brand"
                height={50}
                src="img/logo.jpg"
                width={50}
              />
              More Recipes
            </Link>
          </div>,
          <button
            aria-controls="navbarsExampleDefault"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarsExampleDefault"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>,
          <div
            className="collapse navbar-collapse"
            id="navbarsExampleDefault"
          >
            <ul
              className="navbar-nav mr-auto"
            />
            <form
              className="form-inline my-2 my-md-0"
              onSubmit={[Function]}
            >
              <Field
                className="form-control"
                component="input"
                name="search"
                placeholder="Search"
              />
              <button
                className="btn btn-search my-2 my-sm-0"
                type="submit"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-search"
                />
                 Search
              </button>
            </form>
          </div>,
        ],
        "className": "navbar navbar-expand-md navbar-light",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="nav-link header-logo"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/"
            >
              <img
                alt="logo"
                className="navbar-brand"
                height={50}
                src="img/logo.jpg"
                width={50}
              />
              More Recipes
            </Link>,
            "className": "logo-link-wrap",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className="navbar-brand"
                  height={50}
                  src="img/logo.jpg"
                  width={50}
                />,
                "More Recipes",
              ],
              "className": "nav-link header-logo",
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": "navbar-brand",
                  "height": 50,
                  "src": "img/logo.jpg",
                  "width": 50,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "More Recipes",
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-controls": "navbarsExampleDefault",
            "aria-expanded": "false",
            "aria-label": "Toggle navigation",
            "children": <span
              className="navbar-toggler-icon"
            />,
            "className": "navbar-toggler",
            "data-target": "#navbarsExampleDefault",
            "data-toggle": "collapse",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navbar-toggler-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul
                className="navbar-nav mr-auto"
              />,
              <form
                className="form-inline my-2 my-md-0"
                onSubmit={[Function]}
              >
                <Field
                  className="form-control"
                  component="input"
                  name="search"
                  placeholder="Search"
                />
                <button
                  className="btn btn-search my-2 my-sm-0"
                  type="submit"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-search"
                  />
                   Search
                </button>
              </form>,
            ],
            "className": "collapse navbar-collapse",
            "id": "navbarsExampleDefault",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "navbar-nav mr-auto",
              },
              "ref": null,
              "rendered": null,
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Field
                    className="form-control"
                    component="input"
                    name="search"
                    placeholder="Search"
                  />,
                  <button
                    className="btn btn-search my-2 my-sm-0"
                    type="submit"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-search"
                    />
                     Search
                  </button>,
                ],
                "className": "form-inline my-2 my-md-0",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "form-control",
                    "component": "input",
                    "name": "search",
                    "placeholder": "Search",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        aria-hidden="true"
                        className="fa fa-search"
                      />,
                      " Search",
                    ],
                    "className": "btn btn-search my-2 my-sm-0",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-hidden": "true",
                        "className": "fa fa-search",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " Search",
                  ],
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
